packages/core/admin/admin/src/pages/Home/HomePage.tsx

```diff
@@ -55,14 +57,88 @@ export const WidgetRoot = ({ title, icon = PuzzlePiece, children }: RootProps) =
 };
 
 /* -------------------------------------------------------------------------------------------------
- * HomePageCE
+ * UnstableHomePageCe
  * -----------------------------------------------------------------------------------------------*/
 
+const UnstableHomePageCe = () => {
+  const { formatMessage } = useIntl();
+  const user = useAuth('HomePageCE', (state) => state.user);
+  const displayName = user?.firstname ?? user?.username ?? user?.email;
+  const getAllWidgets = useStrapiApp('UnstableHomepageCe', (state) => state.widgets.getAll);
+
+  const [loadedWidgets, setLoadedWidgets] = React.useState<
+    (Omit<ReturnType<typeof getAllWidgets>[number], 'component'> & {
+      component: React.ComponentType;
+    })[]
+  >([]);
+
+  React.useEffect(() => {
+    const loadComponents = async () => {
+      const widgets = getAllWidgets();
+
+      const loaded = await Promise.all(
+        widgets.map(async (widget) => {
+          const LazyComponent = await widget.component();
+
+          return {
+            ...widget,
+            component: LazyComponent,
+          };
+        })
+      );
+
+      setLoadedWidgets(loaded);
```

Comment:
hmm thinking now I should just lazy load the components I think this blocks the widgets from rendering things like their title until the component is ready

Reply:
nvm it doesn't block anything